### DISCUSSION TITLE
Creating a popup dictionary for other languages

### DISCUSSION THREAD

**Discussion by Boy19991** [melink14/rikaikun#1051] [General]
Greetings,

I'm sure everyone here knows the great power of rikaikun and similiar dictionaries.

I have searched for popup dictionaries in other languages but I havent found anything.

The languages I personally dream of would be Spanish and Hebrew, and then Russian.

Slava Russian dictionary is available on Chrome, but it uses ajax to connect to the wikitionary API, which is slow and interupts your reading flow.

Rikaikun stores the dictionary locally, so you can power through documents quickly and not have to wait for an ajax request.

I am a junior programmer and would probably need someone to provide guidance.

Is there such a super-human persona lurking in these directories?

We will soon see.

Nick.

**Comment by melink14** [comment]
Hi Nick,

I've seen similar pop up dictionaries for chinese but not sure about the other languages!

A general pop up dictionary would be interesting though I'm sure it would require a bit of work up front to make work.

The key problem is source of dictionaries! One reason why this is able to work in Japanese is because for the last 30+ years there has been a community of people contributing and editing a free dictionary database for Japanese -> English.

So the first step for any project like this would be to research digitial dictionaries that were free to use (or could be obtained by users) and then start thinking about how to use them in a popup dictionary. (For example, if you can download wikitionary data then you could probably convert to an easy to use format and install into a popup dictionary)

From a development point of view, I don't have much time to invest in something like that though as just one language keeps my hands full! Definitely possible though!

**Comment by seth-js** [comment]
Wiktionary dumps are available at [kaikki.org](https://kaikki.org/). The data is pretty easy to clean up and make it work with Yomichan. I've already released a couple projects with it. You need to rewrite parts of Yomichan to handle a language with spaces, letter case, and to automatically look up inflected versions of a word and provide inflection information when necessary.

https://github.com/seth-js/yomichan-ru
https://github.com/seth-js/yomichan-de

The Wiktionary dump is also super useful for creating a parser. I've been able to create frequency lists with it based on Netflix subtitle collections I ripped.

**Reply by SakiDailene** [reply]
Hey, can you make a more comprehensive guide on this? Similar to your tidy-kaikki-russian repo.

**Reply by SakiDailene** [reply]
I've got an email about your comment but you seem to have deleted it / I just can't see it for some reason. Just wanted to say that it'd be great / I'd be grateful for that.
